data-plane/bindings/adapter/src/adapter.rs

```diff
@@ -0,0 +1,1841 @@
+        }
+    }
+}
+
+/// Session type enum
+#[derive(uniffi::Enum)]
+pub enum SessionType {
+    PointToPoint,
+    Multicast,
+}
+
+/// Session configuration
+#[derive(uniffi::Record)]
+pub struct SessionConfig {
+    /// Session type (PointToPoint or Multicast)
+    pub session_type: SessionType,
+
+    /// Enable MLS encryption for this session
+    pub enable_mls: bool,
+
+    /// Maximum number of retries for message transmission (None = use default)
+    pub max_retries: Option<u32>,
+
+    /// Interval between retries in milliseconds (None = use default)
+    pub interval_ms: Option<u64>,
+
+    /// Whether this endpoint is the session initiator
+    pub initiator: bool,
+
+    /// Custom metadata key-value pairs for the session
+    pub metadata: std::collections::HashMap<String, String>,
+}
+
+impl From<SessionConfig> for SlimSessionConfig {
+    fn from(config: SessionConfig) -> Self {
+        SlimSessionConfig {
+            session_type: match config.session_type {
+                SessionType::PointToPoint => ProtoSessionType::PointToPoint,
+                SessionType::Multicast => ProtoSessionType::Multicast,
+            },
+            max_retries: config.max_retries,
+            interval: config.interval_ms.map(std::time::Duration::from_millis),
+            mls_enabled: config.enable_mls,
+            initiator: config.initiator,
+            metadata: config.metadata,
+        }
+    }
+}
+
+/// Error types for SLIM operations
+#[derive(Debug, thiserror::Error, uniffi::Error)]
+pub enum SlimError {
+    #[error("Configuration error: {message}")]
+    ConfigError { message: String },
+    #[error("Session error: {message}")]
+    SessionError { message: String },
+    #[error("Receive error: {message}")]
+    ReceiveError { message: String },
+    #[error("Send error: {message}")]
+    SendError { message: String },
+    #[error("Authentication error: {message}")]
+    AuthError { message: String },
+    #[error("Operation timed out")]
+    Timeout,
+    #[error("Invalid argument: {message}")]
+    InvalidArgument { message: String },
+    #[error("Internal error: {message}")]
+    InternalError { message: String },
+}
+
+impl From<ServiceError> for SlimError {
+    fn from(err: ServiceError) -> Self {
+        match err {
+            ServiceError::ConfigError(msg) => SlimError::ConfigError { message: msg },
+            ServiceError::ReceiveError(msg) => SlimError::ReceiveError { message: msg },
+            ServiceError::SessionError(msg) => SlimError::SessionError { message: msg },
+            _ => SlimError::InternalError {
+                message: err.to_string(),
+            },
+        }
+    }
+}
+
+impl From<SlimSessionError> for SlimError {
+    fn from(err: SlimSessionError) -> Self {
+        SlimError::SessionError {
+            message: err.to_string(),
+        }
+    }
+}
+
+/// TLS configuration for server/client
+#[derive(uniffi::Record)]
+pub struct TlsConfig {
+    pub insecure: bool,
```

Comment:
We are missing many parameters for the tls configuration. As follow up, we need to think if there is a clever way to expose all the fields in the configuration structs without redefining everything 🤔